### DISCUSSION TITLE
Uploads succeeding but mc throws error.

### DISCUSSION THREAD

**Discussion by ScionOfDesign** [minio/minio#14008] [Q&A] [answered]
I am configuring our CI/CD server to upload artifacts to Minio but am encountering a somewhat strange issue. The artifacts get uploaded properly but mc ends up throwing an error most of the time.

> <ERROR> Failed to copy `MyArtifact.zip`. The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.

Sometimes it will through a 502 error but still succeed.
```
<html>
<body>
<!--StartFragment-->

mc.exe: <ERROR> Failed to copy `MyArtifact.zip`. <html>
--
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx/1.19.2</center>
</body>
</html>

<!--EndFragment-->
</body>
</html>
```

The logs spit out these errors:

> Error: Unable to update replication metadata for releases/Artifacts/MyArtifact.zip(ef162c12-69c9-4892-b7f5-56a1beeeb8d4): Operation timed out (*fmt.wrapError)
> cmd/bucket-replication.go:933:cmd.replicateObject()
> cmd/bucket-replication.go:1645:cmd.scheduleReplication()
> cmd/object-handlers.go:3302:cmd.objectAPIHandlers.CompleteMultipartUploadHandler()
> net/http/server.go:2047:http.HandlerFunc.ServeHTTP()

> Error: failed to get lock for object: Artifacts/MyArtifact.zip bucket:releases arn: (*errors.errorString)
> cmd/bucket-replication.go:874:cmd.replicateObject()
> cmd/bucket-replication.go:1350:cmd.(*ReplicationPool).AddWorker()

My current setup is that I have two minio clusters on separate servers. I am replicating our artifacts bucket synchronously across both of them. I am accessing the servers through an NGINX reverse proxy. If anyone could provide some help in debugging this issue, it would be greatly appreciated.

**Comment by harshavardhana** [comment, marked as answer]
@ScionOfDesign what is your server version?

**Reply by ScionOfDesign** [reply]
> @harshavardhana I am using the latest Docker image.

Edit: I just updated to the latest again, I will see if it happens again.
Seems to be working now!
